google/cloud/pubsub/internal/tracing_batch_callback.cc

```diff
@@ -150,13 +152,48 @@ class TracingBatchCallback : public BatchCallback {
   }
 
  private:
+  struct MessageSpans {
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> subscribe_span;
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>
+        concurrency_control_span;
+  };
+
+  void StartConcurrencyControl(std::string const& ack_id) override {
+    namespace sc = opentelemetry::trace::SemanticConventions;
+    std::lock_guard<std::mutex> lk(mu_);
+    auto spans = spans_by_ack_id_.find(ack_id);
+    if (spans != spans_by_ack_id_.end()) {
+      auto subscribe_span = spans->second.subscribe_span;
+      if (subscribe_span) {
+        opentelemetry::trace::StartSpanOptions options;
+        options.parent = subscribe_span->GetContext();
+        auto span =
+            internal::MakeSpan("subscriber concurrency control",
+                               {{sc::kMessagingSystem, "gcp_pubsub"}}, options);
+        auto scope = internal::OTelScope(span);
```

Comment:
It looks strange to not do anything except assign a pointer within a scope.

Reply:
Yeah that's what all the tracing is. It's based off of how Otel handles scopes.

Reply:
It keeps the span alive I think?

@dbolduc might have a better explaination

Reply:
But @dbolduc asked the question.

Reply:
Oh I see now.. oops.

Reply:
https://github.com/googleapis/google-cloud-cpp/pull/13913 Addressed here to remove it.